x-element.js

```diff
@@ -765,19 +804,22 @@ export default class XElement extends HTMLElement {
 
   static #initializeHost(host) {
     const hostInfo = XElement.#hosts.get(host);
-    const { computeMap, initialized, invalidProperties } = hostInfo;
+    const { computeMap, initialized, invalidProperties, valueMap } = hostInfo;
     if (initialized === false) {
-      XElement.#upgradeOwnProperties(host);
       // Only reflect attributes when the element is connected.
       const { propertyMap } = XElement.#constructors.get(host.constructor);
       for (const property of propertyMap.values()) {
         const { value, found } = XElement.#getPreUpgradePropertyValue(host, property);
-        XElement.#initializeProperty(host, property);
         if (found) {
-          host[property.key] = property.default(host, property.initial(value));
```

Comment:
There were a couple options here:

1. Introduce a new flag like `initializing` (versus `initialized`) which could help us internally “do the right thing” when we _set_ a value on the host…
2. Use specific, internal methods during initialization because we accept that we are in a strange half-way state and it doesn’t really make sense to be using the full functionality _just yet_.

I went with (2) because the code felt cleaner overall. Additionally, by doing this, I was actually able to _remove _ an internal fork within `#invalidateProperty` 🤙